### DISCUSSION TITLE
Stream Deck: Change the image on buttons from an external program.

### DISCUSSION THREAD

**Discussion by runnowll** [elgatosf/elgato#34] [Help] [answered]
Is there an easy way to change or update different background images of each button? I am running a Windows PC and the application can run on the same PC as the Stream Deck. I can write code in Python, vbscript, visual studio, batch files, power shell, javascript,  and willing to learn more.

As a slide show plays on a PC I want a multi function button to update with the same image. (I create smaller thumbnails)

The closest I have to a solution is to replace the image file that the Stream Deck is using and then refresh the profile. I have a program that replaces the image found in the roaming folder and then switches between two identical profiles. I switch profiles by changing the focused application feature with in Stream Decks configuration. This works but it is messy. Any help you can give is appreciated.

**Comment by tiptronic** [comment, marked as answer]
hi @runnowll 

a pretty simple way would be:

- You only need to install/use 1 profile / per devicetype
- add an empty action for every key of the profile (<- this just receives the image)
- - so you need to prepare some profiles (SD regular, SD mini, SD XL, SD+), depending on which devices you want to support
- - if the user switches to this profile, your plugin is considered 'active'

(You only need the preparation above while creating your plugin - the profiles will be installed by SD, when you install the plugin)

In your plugin's code (easiest to do in Javascript using an HTML-Canvas, imo):
1. get the image from your application (either via file or e.g. websockets)
2. cut it into square tiles (best is a multiple of 72x72px)
3. send it to the individual keys using the `setImage` API
- if the image changes, repeat those 3 steps
- Done

If the user switches to another profile don't do anything (to save cpu).
Just as a note: you can't change or update the `background image` of a key, but the image of the key (but I guess, this i what you meant anyway)
